deepinv/tests/test_datasets.py

```diff
@@ -202,8 +203,111 @@ def test_base_dataset():
             check_dataset(MyDataset(bad_dataset_input))
 
 
+SPLIT_NAMES = ["train", "test", "val", "dummy"]
+
+
+@pytest.mark.parametrize("train", [True, False])
+@pytest.mark.parametrize("split", [*SPLIT_NAMES, None])
+@pytest.mark.parametrize("with_transform", [False, True])
+@pytest.mark.parametrize("load_physics_generator_params", [True, False])
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
+@pytest.mark.parametrize("complex_dtype", [torch.complex64, torch.complex128])
+@pytest.mark.parametrize("length", [10])
+@pytest.mark.parametrize("with_params", [False, True])
+@pytest.mark.parametrize("unsupervised", [False, True])
+@pytest.mark.parametrize("close", [False, True])
+@pytest.mark.parametrize("stack_size", [1, 2, 3])
+def test_hdf5dataset(
+    tmpdir,
+    train,
+    split,
+    with_transform,
+    load_physics_generator_params,
+    dtype,
+    complex_dtype,
+    length,
+    with_params,
+    unsupervised,
+    close,
+    stack_size,
+):
+    path = f"{tmpdir}/dummy.h5"
+
+    if with_transform:
+        transform = MagicMock(side_effect=lambda x: x)
+    else:
+        transform = None
+
+    # Populate a HDF5 file
+    with h5py.File(path, "w") as f:
+        # The stacked attribute is expected to be present only if stack_size > 1,
+        # even though stacked = 1 could be meaningful.
+        if stack_size > 1:
+            f.attrs["stacked"] = stack_size
+
+        def populate_dummy_data(field_name):
+            data = np.random.randn(length, 1, 4, 4)
```

Comment:
Do you think we could make the data fixed and different for train (x, y, kernel) and test (x, y, kernel) so that we can test the right split is returned on the right occassion, and x and y and kernel aren't mixed up?